lib/faraday_middleware/gzip.rb

```diff
@@ -48,7 +48,17 @@ def uncompress_gzip(body)
     end
 
     def inflate(body)
+      # Inflate as a DEFLATE (RFC 1950+RFC 1951) stream
       Zlib::Inflate.inflate(body)
+    rescue Zlib::DataError
```

Comment:
I don't think so.  As described in the Wikipedia page above, there are two kinds of implementations for a single Content-Encoding type `deflate`.